components/Header/Header.tsx

```diff
@@ -0,0 +1,124 @@
+import Image from 'next/image';
+import React, { useState, useEffect } from 'react';
+import { Menu, X } from 'react-feather';
+
+type MenuItem = {
+  text: string;
+  path?: string;
+};
+const defaultMenuItems: MenuItem[] = [
+  {
+    text: 'About us'
+  },
+  {
+    text: 'Press'
+  },
+  {
+    text: 'Careers'
+  },
+  {
+    text: 'Legal'
+  },
+  {
+    text: 'Support'
+  },
+  {
+    text: 'Contact'
+  },
+  {
+    text: 'Sitemap'
+  },
+  {
+    text: 'Cookie settings'
+  }
+];
+
+type TopMenuItem = {
+  text: string;
+  path?: string;
+};
+type HeaderParams = {
+  imageUrl?: string;
+  title?: string;
+  topMenuItems?: TopMenuItem[];
+  menuItems?: MenuItem[];
+};
+
+const defaultTopMenuItems: TopMenuItem[] = [
+  {
+    text: 'Home'
+  },
+  {
+    text: 'Products'
+  },
+  {
+    text: 'Resource'
+  },
+  {
+    text: 'Pricing'
+  }
+];
+export default function Header({
+  imageUrl = 'https://prototyprio.gumlet.io/strapi/20b5e1615519f85953d636ec6a5005ae.png?w=3840&q=75&format=webp&compress=true&dpr=2',
+  title = 'Untitled UI',
+  topMenuItems = defaultTopMenuItems,
+  menuItems = defaultMenuItems
+}: HeaderParams) {
+  const [isOpen, setIsOpen] = useState(true);
+
+  useEffect(() => {
+    if (typeof window !== 'undefined' && window.innerWidth < 768) {
```

Comment:
bu condition'a güzel bir naming yapabiliriz. 
örneğin: 
isClientInMobileView = typeof window !== 'undefined' && window.innerWidth < 768

if (isClientInMobileView)